src/Umbraco.Web/Editors/EntityController.cs

```diff
@@ -250,6 +308,64 @@ public IEnumerable<EntityBasic> GetByIds([FromUri]int[] ids, UmbracoEntityTypes
             return GetResultForIds(ids, type);
         }
 
+        /// <summary>
+        /// Get entities by GUID ids
+        /// </summary>
+        /// <param name="ids"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// We allow for POST because there could be quite a lot of Ids
+        /// </remarks>
+        [HttpGet]
+        [HttpPost]
+        public IEnumerable<EntityBasic> GetByIds([FromJsonPath]Guid[] ids, UmbracoEntityTypes type)
+        {
+            if (ids == null)
+            {
+                throw new HttpResponseException(HttpStatusCode.NotFound);
+            }
+            return GetResultForKeys(ids, type);
+        }
+
+        /// <summary>
+        /// Get entities by string ids - will try to convert to the correct id type (int, guid, udi)
+        /// </summary>
+        /// <param name="ids"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// We allow for POST because there could be quite a lot of Ids
+        /// </remarks>
+        [HttpGet]
+        [HttpPost]
+        public IEnumerable<EntityBasic> GetByIds([FromJsonPath]Udi[] ids, [FromUri]UmbracoEntityTypes type)
+        {
+            if (ids == null)
+            {
+                throw new HttpResponseException(HttpStatusCode.NotFound);
+            }
+
+            if (ids.Length == 0)

```

Comment:
It does have this check in the underlying `GetResultForKeys` and `GetResultForIds` methods